packages/orm/src/collections/OneToOneReference.ts

```diff
@@ -131,12 +133,23 @@ export class OneToOneReference<T extends Entity, U extends Entity> extends Abstr
   }
 
   onEntityDelete(): void {
-    // if (this.isCascadeDelete) {
-    //   this.current({ withDeleted: true }).forEach(getEm(this.entity).delete);
-    // }
+    if (this.isCascadeDelete && this.loaded) {
+      getEm(this.entity).delete(this.loaded);
+    }
+  }
+
+  current(opts?: { withDeleted: boolean; }): U | undefined {
+    return this.filterDeleted(this.loaded, opts);
   }
 
-  async onEntityDeletedAndFlushing(): Promise<void> {}
+  async onEntityDeletedAndFlushing(): Promise<void> {
+    if (this.isCascadeDelete) {
```

Comment:
@zgavin yeah, I was VC'ing with Lucy to handle this case / do this approach, and nudged her towards this.

Understood what you describe is historically what `onEntityDeletedAndFlushign` is for, but seems like, due to the potentially-un-loaded-ness of a o2o, we need an async-capable place to do the cascade deleting.

What is your proposed alternative? 

...looking at `OneToMany`, I see it's `onEntityDelete` "works" when it's unloaded, but only cascades to `addedBeforeLoaded`. Do you remember if it's a known issue that this wouldn't cascade delete to children that are in the not-yet-loaded-`loaded` array, if it were to be loaded?

Reply:
`onEntityDelete` is actually called twice in the life-cycle of a delete operation.   First it's called on the initial `em.delete` call.  This is intended to be synchronous, since `em.delete` is also synchronous.  This is fine, though, because the only objective of this call is to cascade the delete into objects already in the UoW.   The second call is in a `beforeDelete` hook that has itself as a populate hint.  Again, it's fine for this to be sync since the relation should now be loaded. So, there shouldn't really be an issue with the potentially unloadedness of a o2o or any need for an async capable cascade delete code path.

The only caveat here is that o2o might want to search the UoW for an FK pointing to it when it isn't yet loaded to be absolutely sure that it's not already in memory.

Reply:
Ah wow, I'd completely missed that `onEntityDelete` would be called twice...

Lucy, per ^ you should be able to delete `onEntityDeletedAndFlushing` and still have your tests pass.

If you don't mind, would be great to also distill Zach's point ^ into a comment on `onDeleteEntity`, i.e.:

```
// Note this will be called twice, once on em.delete (may not be loaded), and again from
// the `beforeFlush` hook that `cascadeDelete` adds (and will be loaded).
onEntityDelete(): void {
   ...
}
```

Reply:
> If i'm understanding your last paragraph, then yes it's a known issue. 

Cool ... wonder if we should how persist the cascade-deleted-ness of `someEntity.someCascadeCollection` and then pre-emptively `em.delete` them on that `.load` call. Not a big deal / would be a separate PR from this one.

Reply:
> Lucy, per ^ you should be able to delete onEntityDeletedAndFlushing and still have your tests pass.

That is not actually what I was intending.  The tests may pass, but the behavior is certainly not correct.  `onEntityDeletedAndFlushing` still needs to have an implementation because the relation and entities involved still need to be cleaned up.  It just shouldn't be calling `em.delete`.  I pointed to `OneToMany` as an example because it's pretty much needs the same thing, just instead of an array of `loaded` entities, it's just a single entity.

> wonder if we should how persist the cascade-deleted-ness of someEntity.someCascadeCollection and then pre-emptively em.delete them on that .load call

Reply:
> wonder if we should how persist the cascade-deleted-ness of someEntity.someCascadeCollection and then pre-emptively em.delete

In practical terms, i'm not sure it's necessary. Anywhere we are calling `em.delete` we are either immediately calling `em.flush` or we've already pre-loaded all our entities/relations.  This could also result in a significant performance regression because it's not really sufficient to only do this check in the relation.  You'd need to check if an entity should be cascade deleted every time you instantiate one or change `isEntityDeleted` cascade aware, which is a problem since it's called a lot and it'd need to search through the UoW.